### PR TITLE
[2007] Validate that a course has sites before publishing

### PR DESCRIPTION
**Pull request by tvararu**
### Context

We shouldn't allow publishing courses without at least one site

### Changes proposed in this pull request

Add validation with message, update specs

### Guidance to review

:shipit: 

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by tvararu** [comment]
[Fix issue with rolled over 2020 courses with 0 `site_statuses`](https://trello.com/c/WiXnwzp8/2007-fix-issue-with-rolled-over-2020-courses-with-0-sitestatuses)

File: spec/models/course/publishable_spec.rb [hunk 1]
```diff
@@ -9,15 +11,33 @@
...
+        end
+
+        it { should_not be_empty }
+      end
+    end
+
+    context 'with no sites' do
+      let(:enrichment) { build(:course_enrichment, :subsequent_draft, created_at: 1.day.ago) }
+      let(:course) {
+        create(:course, enrichments: [enrichment])
```
Comment (richrace) [inline spec/models/course/publishable_spec.rb]: Should we explicitly set `site_statuses` as empty here?

```suggestion
        create(:course, site_statuses: [], enrichments: [enrichment])
```

File: spec/requests/api/v2/providers/courses/publish_spec.rb [hunk 2]
```diff
@@ -109,18 +109,24 @@
     describe 'failed validation' do
       let(:json_data) { JSON.parse(subject.body)['errors'] }
 
-      context 'no enrichments' do
+      context 'no enrichments or sites' do
         let(:course) { create(:course, provider: provider) }
```
Comment (richrace) [inline spec/requests/api/v2/providers/courses/publish_spec.rb]: And add the similar explicit no site/enrichment here too.

File: spec/requests/api/v2/providers/courses/publish_spec.rb [hunk 3]
```diff
@@ -109,18 +109,24 @@
     describe 'failed validation' do
       let(:json_data) { JSON.parse(subject.body)['errors'] }
 
-      context 'no enrichments' do
+      context 'no enrichments or sites' do
```
Comment (defong) [inline spec/requests/api/v2/providers/courses/publish_spec.rb]: can we break this into two tests, please enrichment is painful enough.

Change the original course to fulfil the original tests
then add an explicit one for the no sites.

Comment (richrace) [reply spec/requests/api/v2/providers/courses/publish_spec.rb]: I read it as 'no enrichments AND no sites'

Comment (tvararu) [reply spec/requests/api/v2/providers/courses/publish_spec.rb]: Yeah, I don't think it's necessary to break this up. The file is already quite long...

Comment (defong) [reply spec/requests/api/v2/providers/courses/publish_spec.rb]: cool

**Pull request merged**